crates/storage/src/hyperlane/snapshot.rs

```diff
@@ -11,12 +11,23 @@ use std::sync::{Arc, RwLock};
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct HyperlaneSnapshot {
+    // the trusted EV height in the ZKISM
     pub height: u64,
+    // the Hyperlane Message Tree e.g. Snapshot
     pub tree: MerkleTree,
+    // whether this Snapshot has been finalized
+    pub finalized: bool,
 }
 impl HyperlaneSnapshot {
     pub fn new(height: u64, tree: MerkleTree) -> HyperlaneSnapshot {
-        HyperlaneSnapshot { height, tree }
+        HyperlaneSnapshot {
+            height,
+            tree,
+            finalized: false,
+        }
+    }
+    pub fn finalize(&mut self) {
+        self.finalized = true;
```

Comment:
would it help to verify that finalized is set to false? to ensure certain invariant in the code?